### DISCUSSION TITLE
Run pyright in single file mode and some questions

### DISCUSSION THREAD

**Discussion by f4T1H21** [microsoft/pyright#7096] [General]
Hello,
I have pyright set up as an lsp in neovim and my configuration is like this:

```lua
lspconfig.pyright.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  root_dir = function(fname)
    local root_files = {
      "pyproject.toml",
      "setup.py",
      "setup.cfg",
      "requirements.txt",
      "Pipfile",
      "pyrightconfig.json",
    }
    return util.root_pattern(table.unpack(root_files))(fname)
        or util.find_git_ancestor(fname)
        or util.path.dirname(fname)
  end,
  flags = { debounce_text_changes = 300 },
  settings = {
    python = {
      analysis = {
        autoSearchPaths = true,
        diagnosticMode = "openFilesOnly",
        useLibraryCodeForTypes = true,
        typeCheckingMode = "basic",
      },
    },
  },
})

```

I have some questions regarding the configuration above:

1- When I set `root_dir = util.root_pattern(vim.loop.cwd())` the `root_dir` in LspInfo is always set as `/` even if my current working directory is somewhere else. Why does that happen? 

2- What's the function of `debounce_text_changes = 300` flag?

3- In this configuration, it sets the `root_dir` current working directory if there's no project file found. However I want it to run in `single file mode` when there are no project files/folders (eg .git/setup.py) exist. How do I accomplish that?

4- What does `autoSearchPaths = true` do?

5- What does using `LibraryCode` for `Types` mean? What is `LibraryCode` and what are `Types`?

Thank you in advance!
All the best...

**Comment by erictraut** [comment]
You will probably need to post your questions elsewhere to get the answers you're looking for. This is the project site for pyright, but your questions are related to neovim and its LSP support. I'm the primary maintainer of pyright, but I have little or no knowledge of neovim.

**Comment by f4T1H21** [comment]
The questions are answered [here](https://vi.stackexchange.com/questions/43938/run-pyright-in-single-file-mode-and-some-questions) properly, thanks.
